client/src/features/modals/AppSettingsModal.jsx

```diff
@@ -95,33 +105,27 @@ export default function AppSettingsModal() {
           e.status = true;
           e.message += 'App pin code added';
         }
+      }
 
-        // set fields with error
-        if (!e.status) {
-          emitError(`Invalid Input: ${e.message}`);
-        } else {
-          await postSettings(formData);
-          await refetch();
-          emitWarning(e.message);
-          setChanged(false);
+      if (f.timeFormat !== data.timeFormat) {
+        if (f.timeFormat === '12' || f.timeFormat === '24') {
+          e.status = true;
         }
       }
+
+      // set fields with error
+      if (!e.status) {
+        emitError(`Invalid Input: ${e.message}`);
+      } else {
+        await postSettings(formData);
+        await refetch();
+        emitWarning(e.message);
+        setChanged(false);
+      }
       setSubmitting(false);
       setChanged(false);
     },
-    [
-      data.pinCode,
-      doDefaultPublic,
-      doShowQuickEntry,
-      doStarTimeIsLastEnd,
-      emitError,
-      emitWarning,
-      formData,
-      refetch,
-      setDefaultPublic,
-      setShowQuickEntry,
-      setStarTimeIsLastEnd,
-    ]
+    [data.pinCode, data.timeFormat, doDefaultPublic, doShowQuickEntry, doStarTimeIsLastEnd, emitError, emitWarning, formData, refetch, setDefaultPublic, setShowQuickEntry, setStarTimeIsLastEnd]
```

Comment:
There is no need for wrapping this event handler in a `useCallback` because you're passing it to an HTML element, therefore it will not be expensive if it needs to "rerender".

I guess in this case the complexity is not worth the gains.

Reply:
The same goes for the other event handlers in this component.

`handleContextChange` for example, is being called inside an arrow function that is re-created on every render. Therefore having `handleContextChange` inside a `useCallback` does nothing. The same goes for `handleChange`.

The `revert` is passed to a component that re-renders if this is not a stable reference, but I think the component is so simple it makes no difference performance-wise.